### PR TITLE
feat: add tool PR comments

### PR DESCRIPTION
**Pull request by sqin2019**
part of #64

### CODE REVIEW

**Comment by yolocs** [comment]
Do you have a screenshot of how the comment would look like?

**Comment by sqin2019** [comment]
> Do you have a screenshot of how the comment would look like?

Screenshot 2023-07-19 at 4 43 50 PM
Screenshot 2023-07-19 at 4 43 28 PM

File: .github/workflows/handle.yml [hunk 1]
```diff
@@ -127,17 +128,16 @@ jobs:
...
-            aod tool do -path ${{ env.TOOL_FILE_PATH }}
-          else
-            echo "::notice title=Tool Request Handle::Skip because tool.yaml is not found"
-          fi
+          touch ${{ env.TOOL_ERROR_FILENAME }}
+          aod tool do -path ${{ env.TOOL_FILE_PATH }} 2> ${{ env.TOOL_ERROR_FILENAME }}
 
       - name: 'IAM Request Comment'
-        if: '${{ always()}}'
+        if: '${{ always() && hashFiles(''iam.yaml'') != '''' }}'
```
Comment (yolocs) [inline .github/workflows/handle.yml:140]: Help me understand - is `always()` useless here?

File: .github/workflows/handle.yml [hunk 2]
```diff
@@ -191,13 +191,99 @@ jobs:
...
+                req = fs.readFileSync(
+                  `tool.yaml`,
+                  { encoding: "utf8" }
+                );
+
+              body = `**\`Access on Demand\`** - 🟩 **\`Tool\`** request succeeded.
+
+            <details>
+            <summary>Details</summary>
+            Executed "do" commands in below request.
```
Comment (yolocs) [inline .github/workflows/handle.yml:228]: I think the correct gramma is "...in the request below"

File: .github/workflows/handle.yml (hunk 1 above)
Comment (sqin2019) [reply .github/workflows/handle.yml:140]: No, always means no matter previous steps failed or succeed or skipped, without this, default is all previous steps must succeed.

File: .github/workflows/handle.yml [hunk 3]
```diff
@@ -191,13 +191,99 @@ jobs:
...
+            ${req}
+            \`\`\`
+
+            Error:
+            \`\`\`
+            ${error}
+            \`\`\`
+            </details>`;
+                break;
+              // step cancelled.
```
[the remark this answers is not in the document]
Comment (sqin2019) [reply .github/workflows/handle.yml:260]: When the step is cancelled/skipped.

File: .github/workflows/handle.yml [hunk 4]
```diff
@@ -191,13 +191,99 @@ jobs:
...
+        with:
+          github-token: '${{ github.token }}'
+          retries: '3'
+          script: |+
             await github.rest.issues.createComment({
               owner: context.repo.owner,
               repo: context.repo.repo,
               issue_number: ${{ github.event.pull_request.number }},
-              body: body,
+              body: `**\`Access on Demand\`** - 🟦 request not found.
```
Comment (sqin2019) [inline .github/workflows/handle.yml:286]: Changed to blue to match with https://github.com/abcxyz/guardian/pull/148#issuecomment-1644477772

**Review by yolocs — APPROVED** [review]

**Pull request merged**